### PR TITLE
onewire-list command added

### PR DESCRIPTION
**Pull request by libesz**
I created a simple and generic onewire device listing command. I found it helpful to test host implementations.

### CODE REVIEW

File: cmd/onewire-list/main.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Onewire-list lists all onewire buses and devices. It furthermore treats
```
Comment (maruel) [inline cmd/onewire-list/main.go]: It's conflating two separate tasks: one to list all the devices detected on onewire bus, the other to query a ds18b20. I think I'd prefer two single-purpose focused executables instead.

Comment (libesz) [reply cmd/onewire-list/main.go]: Okay. So I removed the device specific part from this.

File: cmd/onewire-list/main.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	"periph.io/x/periph/conn/onewire/onewirereg"
+	"periph.io/x/periph/host"
+)
+
+func mainImpl() error {
+	if _, err := host.Init(); err != nil {
+		return err
+	}
+	for _, ref := range onewirereg.All() {
+		fmt.Println("BUS name:", ref.Name)
```
Comment (maruel) [inline cmd/onewire-list/main.go]: Can you make its output closer to https://github.com/google/periph/blob/master/cmd/spi-list/main.go and https://github.com/google/periph/blob/master/cmd/i2c-list/main.go ?

In particular the bus number, and print the Q pin. Obviously the Search() is onewire specific. That said I'm also open to tweak i2c-list and spi-list if you have improvement ideas.

**Comment by maruel** [comment]
gohci

File: cmd/onewire-list/main.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Onewire-list lists all onewire buses and devices.
+
```
[the remark this answers is not in the document]
Comment (libesz) [reply cmd/onewire-list/main.go]: Fixed.

File: cmd/onewire-list/main.go (hunk 2 above)
Comment (libesz) [reply cmd/onewire-list/main.go]: I tried to align with the other commands. I have no improvement idea in general.

**Comment by maruel** [comment]
Change looks good to me, can you confirm you signed the CLA before I merge in?

**Comment by libesz** [comment]
I signed it!

**Comment by maruel** [comment]
gohci

**Comment by maruel** [comment]
Reviewing this PR made me realize the tools weren't consistent, so I sent https://github.com/google/periph/pull/208. Please to update the flag processing based on this PR.
You may also (optionally) want to add yourself to AUTHORS and CONTRIBUTORS.

I can't really make sure of the tool yet as I don't have yet a platform with 1-wire enabled but it's not a big deal.

Thanks!

**Comment by maruel** [comment]
I'll just merge it and will rebase my PR instead.

**Comment by libesz** [comment]
Sorry, I had no chance to update the PR yesterday. Thanks for merging anyway!

**Pull request merged**